### DISCUSSION TITLE
Is it possible to write an enumeration value to the OPC UA Server?

### DISCUSSION THREAD

**Discussion by tommy-emerson** [FreeOpcUa/opcua-asyncio#1747] [General]
Hi,
I can write ints, floats and strings to the OPC UA server using the opcua-asyncio's sync wrapper client, by simply calling node.set_value(data_value).
However, I stumbled into problems when writing a custom enumeration, because the server did not like that I tried to write an enum-value of type Int32. At least, that is how I understand the problem. So I looked for a way to specify the Enumeration data type after having extracted it from the node using: enum_data_type_id = node.read_data_type(). Unfortunately it seems that the Variant type does not have a DataType accepting enum_data_type_id, neither does DataValue. So, I do not know how to specify that I'm writing a value of a certain Enumeration type - is it possible? How?
Regards,
TommyS

**Comment by AndreasHeine** [comment]
https://github.com/FreeOpcUa/opcua-asyncio/blob/302c434379490aba8925b9d925de26581a482633/tests/test_common.py#L1199

maybe this helps tbh i have not tried it yet but you could share your sample code!?

**Comment by schroeder-** [comment]
You can do it like this, replace **MyEnum**. We use a python enum class for opc ua enums.
```
....
await client.load_data_type_definitions()
....
await node.write_value(ua.MyEnum.toto)
```
